tests/test_ha.py

```diff
@@ -395,3 +400,11 @@ def test_post_recover(self):
         self.assertEqual(self.ha.post_recover(), 'failed to start postgres')
         self.p.is_running = true
         self.assertIsNone(self.ha.post_recover())
+
+    def test_schedule_future_restart(self):
+        self.ha.patroni.scheduled_restart = {}
+        self.ha.schedule_future_restart({'schedule': '2016-08-30 12:45TZ+1"'})
+        self.ha.schedule_future_restart({'schedule': '2016-08-30 12:45TZ+1"'})
```

Comment:
Actually the reason is that the second restart should fail, because the first one is scheduled.